src/main/java/edu/ie3/datamodel/io/sink/SqlSink.java

```diff
@@ -0,0 +1,521 @@
+                            persistAll(arr, identifier);
+                        } catch (ExtractorException | SQLException e) {
+                            log.error(
+                                    String.format(
+                                            "An error occurred during extraction of nested entity'%s': ",
+                                            cls.getSimpleName()),
+                                    e);
+                        }
+                    }
+            );
+            insertListIgnoreNested(entities, cls, identifier);
+        } else {
+            insertListIgnoreNested(entities, cls, identifier);
+        }
+    }
+
+    public <C extends UniqueEntity> void persistIncludeNested(C entity, DatabaseIdentifier identifier) throws SQLException {
+        Set<C> entitiesToAdd = new HashSet<>();
+        entitiesToAdd.add(entity);
+        persistAll(entitiesToAdd, identifier);
+    }
+
+    private <C extends UniqueEntity> void persistMixedList(List<C> entities, DatabaseIdentifier identifier) {
+        Map<Class<C>, List<C>> entitiesPerClass = entities.stream()
+                .collect(groupingBy(entity -> (Class<C>) entity.getClass()));
+        entitiesPerClass.forEach(
+                (cls, ent) -> {
+                    try {
+                        persistList(ent, cls, identifier);
+                    } catch (SQLException e) {
+                        log.error(String.format(
+                                        "An error occurred during extraction of entity'%s': ",
+                                        cls.getSimpleName()),
+                                e);
+                        throw new RuntimeException(String.format(
+                                "An error occurred during extraction of entity '%s', SQLReason: '%s'",
+                                cls.getSimpleName(), e.getMessage()), e);
+                    }
+                }
+        );
+    }
+
+    private <C extends UniqueEntity, E extends TimeSeriesEntry<V>, V extends Value> void persistList(List<C> entities, Class<C> cls, DatabaseIdentifier identifier) throws SQLException {
+        // Check if there are only elements of the same class
+        Class<?> firstClass = entities.get(0).getClass();
+        boolean allSameClass = true;
+        for (Object obj : entities) {
+            if (obj.getClass() != firstClass) {
+                allSameClass = false;
+                break;
+            }
+        }
+
+        if (allSameClass) {
+            if (InputEntity.class.isAssignableFrom(cls)) {
+                insertListIgnoreNested(entities, cls, identifier);
+            } else if (ResultEntity.class.isAssignableFrom(cls)) {
+                insertListIgnoreNested(entities, cls, identifier);
+            } else if (TimeSeries.class.isAssignableFrom(cls)) {
+                persistListOfTimeSeries((List<TimeSeries<E, V>>) entities, identifier);
```

Comment:
We can get rid of the method as it reduces to the following one-liner:

```suggestion
                entities.forEach(ts -> persistTimeSeries((TimeSeries<E, V>) ts, identifier));
```